contrib/generic_proxy/filters/network/source/router/router.cc

```diff
@@ -212,6 +212,15 @@ void UpstreamRequest::onBindSuccess(Network::ClientConnection& conn,
   parent_.request_encoder_->encode(*parent_.request_, *this);
 }
 
+void UpstreamRequest::onDecodingSuccess(ResponsePtr response, ExtendedOptions options,
+                                        bool end_stream) {
+  if (end_stream) {
+    clearStream(options.drainClose());
+  }
+
+  parent_.onUpstreamResponse(std::move(response), options);
+}
```

Comment:
This is basically OK if one upstream connection only serve one request at same time. (No connection sharing, no multiplexing).

Reply:
And note if one encoder filter stop the filter chain, the previous old frame will be overwrite by new frame.